functions/go/set-labels/transformer/labels.go

```diff
@@ -92,28 +92,162 @@ func (p *LabelTransformer) setLabelsInSpecs(o *fn.KubeObject) error {
 
 // Transform updates the labels in the right path using GVK filter and other configurable fields
 func (p *LabelTransformer) Transform(objects fn.KubeObjects) error {
+	if objects.Len() == 0 || objects[0] == nil {
+		return fmt.Errorf("no resources found")
+	}
 	for _, o := range objects.WhereNot(func(o *fn.KubeObject) bool { return o.IsLocalConfig() }) {
 		// this label need to set for all GVK
-		metaLabelsPath := FieldPath{"metadata", "labels"}
-		updatedLabels, err := updateLabels(&o.SubObject, metaLabelsPath, p.NewLabels, true)
-		if err != nil {
+		if err := p.setMetadataForAll(o); err != nil {
 			return err
 		}
-		p.LogResult(o, metaLabelsPath, updatedLabels)
 		// set other common labels according to specific GVK
-		err = p.setLabelsInSpecs(o)
-		if err != nil {
+		if err := p.setCommonSpecLabels(o); err != nil {
+			return err
+		}
+		// handle special cases when slices are involved
+		if err := p.setLabelsInSlice(o); err != nil {
 			return err
 		}
-		// handle special case with slice
-		if o.IsGVK("apps", "", "StatefulSet") {
-			if o.GetMap("spec") != nil {
-				for _, vctObj := range o.GetMap("spec").GetSlice("volumeClaimTemplates") {
-					updatedLabels, err = updateLabels(vctObj, metaLabelsPath, p.NewLabels, true)
+	}
+	return nil
+}
+
+func (p *LabelTransformer) setMetadataForAll(o *fn.KubeObject) error {
+	metaLabelsPath := FieldPath{"metadata", "labels"}
+	updatedLabels, err := updateLabels(&o.SubObject, metaLabelsPath, p.NewLabels, true)
+	if err != nil {
+		return err
+	}
+	p.LogResult(o, metaLabelsPath, updatedLabels)
+	return nil
+}
```

Comment:
Suggest using the `o.SetLabel` or `o.SetLabels` functions directly.

Reply:
I am not sure what you mean here. Do you mean to change the name? `o.SetLabels` is the function that performs all label updates according to `resourceList`. Here I am separating all those updates into 3 parts, and this is part to `handle metadata.labels`.